packages/fxa-auth-db-mysql/lib/db/schema/patch-115-116.sql

```diff
@@ -0,0 +1,75 @@
+--
+-- This migration uses the `JSON_OBJECTAGG` function to aggregate available device commands.
+-- Previously we were using a helper function `aggregateNameValuePairs` introduced in
+-- https://github.com/mozilla/fxa-auth-db-mysql/pull/354
+--
+-- Some things to note in this migration, `JSON_OBJECTAGG` errors if any key is NULL,
+-- which might happen when joining deviceCommands and deviceCommandIdentifiers tables.
```

Comment:
Curious if someone knows a way to do this without having to use `JSON_REMOVE`, I poked around a lot and couldn't find a solution that wasn't ugly.

Reply:
I can't think of one that avoids nulls since it's a left join. a subselect could do the objectagg at a lower level but idk that it'd be better. it'd also return null instead of {}.

```
...
  LEFT JOIN (
    (SELECT dc.uid, dc.deviceId, JSON_OBJECTAGG(ci.commandName, dc.commandData) as deviceAvailableCommands FROM 
    deviceCommands AS dc FORCE INDEX (PRIMARY)
    INNER JOIN deviceCommandIdentifiers AS ci FORCE INDEX (PRIMARY)
      ON ci.commandId = dc.commandId
    WHERE ci.commandName IS NOT NULL
    ) as dac
  ) ON (dac.uid = d.uid AND dac.deviceId = d.id)
...
```

Reply:
Your example works well and allowed me to remove the extra check for tokenData 👍🏽